File: AutomaticBrewingCoffeeFE/interfaces/notification.ts
import { ENotificationType, EReferenceType, ESeverity } from "@/enum/notification";
import { Account } from "./account";

export interface Notification {
    notificationId: string;
    title: string;
    message: string;
    notificationType: ENotificationType;
    referenceId: string;
    referenceType: EReferenceType;
    severity: ESeverity;
    createdBy: string;
    notificationRecipients: NotificationRecipient[];
    createdDate: string;
    isRead: boolean;
}

export interface NotificationRecipient {
    notificationRecipientId: string;
    notificationId: string;
    notification: Omit<Notification, "notificationRecipients">;
    accountRole: string;
    accountId: string;
    account: Account;
    isRead: boolean;
    readDate?: string;
}

File: AutomaticBrewingCoffeeBE/AutomaticBrewingCoffee.Main/Domains/Models/Notification.cs
using System.ComponentModel.DataAnnotations;

namespace AutomaticBrewingCoffee.Domain.Models;

public class Notification : BaseModel
{
    [Key] [StringLength(50)] [Required] public string NotificationId { get; set; } = null!;

    [StringLength(100)] public string Title { get; set; } = null!;

    [StringLength(2048)] public string Message { get; set; } = null!;

    [StringLength(50)] public string NotificationType { get; set; } = null!;

    [StringLength(50)] public string? ReferenceId { get; set; }

    [StringLength(50)] public string? ReferenceType { get; set; }

    [StringLength(50)] public string? Severity { get; set; }

    [StringLength(50)] public string CreatedBy { get; set; } = null!;

    public ICollection<NotificationRecipient> NotificationRecipients { get; set; } = new List<NotificationRecipient>();
}